pandas/core/arrays/categorical.py

```diff
@@ -2494,6 +2494,20 @@ def isin(self, values):
         code_values = code_values[null_mask | (code_values >= 0)]
         return algorithms.isin(self.codes, code_values)
 
+    def replace(self, to_replace, value, inplace=False):
+        inplace = validate_bool_kwarg(inplace, "inplace")
+        cat = self if inplace else self.copy()
+        categories = cat.categories.tolist()
```

Comment:
we shouldn't need to use .tolist() here (very inefficient) as its already an Index

Reply:
fixed, toList is not used unless it is needed